src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -27,160 +72,895 @@ const ActorInit Obj_Bean_InitVars = {
+    this->unk_1E8 = func_80937C24;
+}
+
+void func_80937C24(ObjBean* this) {
+}
+
+void func_80937C30(ObjBean* this) {
+    this->unk_1E8 = func_80937CA0;
+    this->unk_1EC = Rand_S16Offset(12, 40);
+    this->unk_1F0 = Rand_S16Offset(200, 400);
+    this->unk_1F2 = 20;
+    this->unk_1F6 = Rand_S16Offset(100, 800);
+    this->unk_1F8 = 20;
+}
+
+void func_80937CA0(ObjBean* this) {
+    this->unk_1EC--;
+    func_80937B54(this);
+    if (this->unk_1EC < 0) {
+        func_80937CE4(this);
+    }
+}
+
+void func_80937CE4(ObjBean* this) {
+    this->unk_1E8 = func_80937D54;
+    this->unk_1EC = Rand_S16Offset(30, 4);
+    this->unk_1F0 = Rand_S16Offset(2000, 1000);
+    this->unk_1F2 = 200;
+    this->unk_1F6 = Rand_S16Offset(14000, 6000);
+    this->unk_1F8 = 4000;
+}
+
+void func_80937D54(ObjBean* this) {
+    this->unk_1EC--;
+    if (this->unk_1EC == 14) {
+        this->unk_1F0 = Rand_S16Offset(200, 400);
+        this->unk_1F6 = Rand_S16Offset(100, 500);
+        this->unk_1F8 = 2000;
+    }
+
+    func_80937B54(this);
+
+    if (this->unk_1EC < 0) {
+        func_80937C30(this);
+    }
+}
+
+void func_80937DD8(ObjBean* this) {
+    this->actionFunc = func_80937DEC;
+}
+
+void func_80937DEC(ObjBean* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp20 = OBJBEAN_GET_C000(&this->dyna.actor);
+
+    if (sp20 == ENOBJBEAN_GET_C000_1) {
+        s32 pad2;
+
+        if (func_809374F8(this, globalCtx) == NULL) {
+            this->unk_1FF = false;
+        } else {
+            this->unk_1FF = true;
+        }
+    } else {
+        this->unk_1FF = false;
+    }
+
+    if (this->unk_1FF && !Flags_GetSwitch(globalCtx, OBJBEAN_GET_3F80(&this->dyna.actor, 1)) &&
+        Flags_GetSwitch(globalCtx, OBJBEAN_GET_3F80(&this->dyna.actor, 0)) && func_800FE9B4(globalCtx)) {
+        Flags_SetSwitch(globalCtx, OBJBEAN_GET_3F80(&this->dyna.actor, 1));
+    }
+
+    if (this->unk_1FF && Flags_GetSwitch(globalCtx, OBJBEAN_GET_3F80(&this->dyna.actor, 1))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+
+    if (sp20 == ENOBJBEAN_GET_C000_2) {
+        this->dyna.actor.draw = func_80938F50;
+    } else {
+        this->dyna.actor.draw = func_80938E00;
+    }
+
+    if (this->unk_1FF && Flags_GetSwitch(globalCtx, OBJBEAN_GET_3F80(&this->dyna.actor, 0))) {
+        this->unk_1FE |= 5;
+        func_80937C30(this);
+    } else {
+        this->unk_1FE |= 1;
+        func_80937C10(this);
+    }
+
+    if (Flags_GetSwitch(globalCtx, OBJBEAN_GET_7F(&this->dyna.actor, 0))) {
+        this->unk_1E4 = 5;
+    } else {
+        this->unk_1E4 = 0;
+    }
+
+    func_80937FB0(this);
+}
+
+void func_80937FB0(ObjBean* this) {
+    this->unk_1E0 = 0;
+    this->actionFunc = func_80937FC8;
+}
+
+void func_80937FC8(ObjBean* this, GlobalContext* globalCtx) {
+    this->unk_1E8(this);
+
+    if (Actor_ProcessTalkRequest(&this->dyna.actor, &globalCtx->state)) {
+        if (Player_GetExchangeItemId(globalCtx) == 0x2E) {
```

Comment:
Looks like `0x2E` should be part of the ExchangeItemID enum, could you add it (even just a place holder) and use it.